### PR TITLE
Support for working directory and environment set through CMakeLists.txt

### PR DESCRIPTION
**Pull request by jgoshi**

### CODE REVIEW

File: BoostTestAdapter/Utility/BoostTestRunnerCommandLineArgsEx.cs [hunk 1]
```diff
@@ -54,12 +54,34 @@ public static void SetWorkingEnvironment(this BoostTestRunnerCommandLineArgs arg
             // Default working directory
             args.WorkingDirectory = Path.GetDirectoryName(source);
 
+            if (settings.CTestProperties != null)
+            {
+                var normalizedSource = source.Replace('/', '\\');
+
+                foreach (var test in settings.CTestProperties)
```
Comment (rchiodo) [inline BoostTestAdapter/Utility/BoostTestRunnerCommandLineArgsEx.cs]: Not sure I'd call it CTest. These are really more generic right?

**Comment by rchiodo** [comment]
@spebl can you take a look too?

**Comment by rchiodo** [comment]
@spebl is added to the review. #Closed

**Review by rchiodo — DISMISSED** [review]
:shipit:

**Comment by spebl** [comment]
Please update the last edited date on the files that have them, other than that the changes look good to me.

**Review by spebl — APPROVED** [review]

**Pull request merged**